src/endpoints/pool/pool.controller.ts

```diff
@@ -0,0 +1,64 @@
+import { ParseAddressAndMetachainPipe, ParseAddressArrayPipe, ParseIntPipe, ParseTransactionHashPipe } from "@multiversx/sdk-nestjs-common";
+import { Controller, DefaultValuePipe, Get, NotFoundException, Param, Query } from "@nestjs/common";
+import { ApiNotFoundResponse, ApiOkResponse, ApiOperation, ApiQuery, ApiTags } from "@nestjs/swagger";
+import { PoolService } from "./pool.service";
+import { QueryPagination } from "src/common/entities/query.pagination";
+import { TransactionInPool } from "./entities/transaction.in.pool.dto";
+import { TransactionType } from "../transactions/entities/transaction.type";
+import { PoolFilter } from "./entities/pool.filter";
+
+@Controller()
+@ApiTags('pool')
+export class PoolController {
+  constructor(
+    private readonly poolService: PoolService,
+  ) { }
+
+  @Get("/pool")
+  @ApiOperation({ summary: 'Transactions pool', description: 'Returns the transactions that are currently in the memory pool.' })
+  @ApiOkResponse({ type: [TransactionInPool], isArray: true })
+  @ApiQuery({ name: 'from', description: 'Number of items to skip for the result set', required: false })
+  @ApiQuery({ name: 'size', description: 'Number of items to retrieve', required: false })
+  @ApiQuery({ name: 'sender', description: 'Search in transaction pool by a specific sender', required: false })
+  @ApiQuery({ name: 'receiver', description: 'Search in transaction pool by a specific receiver', required: false })
+  @ApiQuery({ name: 'type', description: 'Search in transaction pool by type', required: false })
+  async getTransactionPool(
+    @Query('from', new DefaultValuePipe(0), ParseIntPipe) from: number,
+    @Query('size', new DefaultValuePipe(25), ParseIntPipe) size: number,
+    @Query('sender', ParseAddressAndMetachainPipe) sender?: string,
+    @Query('receiver', ParseAddressArrayPipe) receiver?: string,
+    @Query('type') type?: TransactionType,
+  ): Promise<TransactionInPool[]> {
+    return await this.poolService.getPool(new QueryPagination({ from, size }), new PoolFilter({ sender: sender, receiver: receiver, type: type }));
+  }
+
+  @Get("/pool/count")
+  @ApiOperation({ summary: 'Transactions pool count', description: 'Returns the number of transactions that are currently in the memory pool.' })
+  @ApiOkResponse({ type: Number })
+  @ApiQuery({ name: 'sender', description: 'Returns the number of transactions with a specific sender', required: false })
+  @ApiQuery({ name: 'receiver', description: 'Returns the number of transactions with a specific receiver', required: false })
+  @ApiQuery({ name: 'type', description: 'Returns the number of transactions with a specific type', required: false })
+  async getTransactionPoolCount(
+    @Query('sender', ParseAddressAndMetachainPipe) sender?: string,
+    @Query('receiver', ParseAddressArrayPipe) receiver?: string,
+    @Query('type') type?: TransactionType,
```

Comment:
Use the ParseEnumPipe as following:
@Query('type', new ParseEnumPipe(TransactionType)) type?: TokenType,TransactionType

Reply:
done